php/cache/class-cache-point.php

```diff
@@ -0,0 +1,689 @@
+<?php
+/**
+ * Handles cache point management.
+ *
+ * @package Cloudinary
+ */
+
+namespace Cloudinary\Cache;
+
+use Cloudinary\Cache;
+use Cloudinary\Cache\Cache_Controller;
+
+/**
+ * Class Cache Point.
+ *
+ * Handles managing cache points.
+ */
+class Cache_Point {
+
+	/**
+	 * The plugin instance.
+	 *
+	 * @var Cache
+	 */
+	protected $cache;
+
+	/**
+	 * Holds the list of active cache_points.
+	 *
+	 * @var \WP_Post[]
+	 */
+	protected $active_cache_points = array();
+
+	/**
+	 * Holds the list of registered cache_points.
+	 *
+	 * @var \WP_Post[]
+	 */
+	protected $registered_cache_points = array();
+
+	/**
+	 * Holds the list of cache points requiring meta updates.
+	 *
+	 * @var array
+	 */
+	public $meta_updates = array();
+
+	/**
+	 * Post type.
+	 *
+	 * @var \WP_Post_Type
+	 */
+	protected $post_type;
+
+	/**
+	 * Holds the post type.
+	 */
+	const POST_TYPE_SLUG = 'cloudinary_asset';
+
+	/**
+	 * Cache Point constructor.
+	 *
+	 * @param Cache $cache The plugin ache object.
+	 */
+	public function __construct( Cache $cache ) {
+		$this->cache = $cache;
+		$this->register_post_type();
+		add_filter( 'update_post_metadata', array( $this, 'update_meta' ), 10, 4 );
+		add_filter( 'get_post_metadata', array( $this, 'get_meta' ), 10, 3 );
+		add_filter( 'delete_post_metadata', array( $this, 'delete_meta' ), 10, 4 );
+		add_action( 'shutdown', array( $this, 'meta_updates' ) );
+	}
+
+	/**
+	 * Update our cache point meta data.
+	 *
+	 * @param null|bool $check      The check to allow short circuit of get_metadata.
+	 * @param int       $object_id  The object ID.
+	 * @param string    $meta_key   The meta key.
+	 * @param mixed     $meta_value The meta value.
+	 *
+	 * @return bool|null
+	 */
+	public function update_meta( $check, $object_id, $meta_key, $meta_value ) {
+
+		if ( self::POST_TYPE_SLUG === get_post_type( $object_id ) ) {
+			$check = true;
+			$meta  = $this->get_meta_cache( $object_id );
+			if ( ! isset( $meta[ $meta_key ] ) || $meta_value !== $meta[ $meta_key ] ) {
+				$meta[ $meta_key ] = $meta_value;
+				$check             = $this->set_meta_cache( $object_id, $meta );
+			}
+		}
+
+		return $check;
+	}
+
+	/**
+	 * Delete our cache point meta data.
+	 *
+	 * @param null|bool $check      The check to allow short circuit of get_metadata.
+	 * @param int       $object_id  The object ID.
+	 * @param string    $meta_key   The meta key.
+	 * @param mixed     $meta_value The meta value.
+	 *
+	 * @return bool
+	 */
+	public function delete_meta( $check, $object_id, $meta_key, $meta_value ) {
+
+		if ( self::POST_TYPE_SLUG === get_post_type( $object_id ) ) {
+			$check = false;
```

Comment:
@pereirinha - that is what we want. When we hook in, we take over the meta for this post type. it will return false when it doesn't delete. if we returned the null, it will attempt the DB method. we need to have it shorted out.